app/models/jobstatusdata.rb

```diff
@@ -74,8 +74,9 @@ def extended_data_torque(info)
     self.output_path = info.native[:Output_Path].to_s.split(":").second || info.native[:Output_Path]
 
     output_pathname = Pathname.new(self.output_path).dirname
-    self.terminal_path = OodAppkit.shell.url(path: (output_pathname.writable? ? output_pathname : ENV["HOME"])).to_s
-    self.fs_path = OodAppkit.files.url(path: (output_pathname.writable? ? output_pathname : ENV["HOME"])).to_s
+    self.file_explorer_url = { path: output_pathname }
```

Comment:
In my opinion I think this trick is too smart for its own good (it appears too magical). I actually thought you were assigning this variable as a hash and was confused until I realized you have an uber-custom assignment method.

I think it might have been better to make this a normal private method such as:

```ruby
self.file_explorer_url = get_file_explorer_url(path: output_pathname)
```

and for below:

```ruby
self.shell_url = get_shell_url(path: output_pathname, cluster: self.cluster)
```